src/java/com/github/ruleant/getback_gps/StateTrackingDialogBuilder.java

```diff
@@ -0,0 +1,171 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package com.github.ruleant.getback_gps
+ * @author Jan Scheible
+ */
+package com.github.ruleant.getback_gps;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.StringRes;
+
+/**
+ * Custom dialog builder that uses {@code AlertDialog.Builder} under the hood. The text values
+ * of all {@code TextView} views are automatically synced with the state.
+ * Also standard use cases as the negative button or dialog dismiss are covered and clear the state.
+ */
+class StateTrackingDialogBuilder {
+
+    private final AlertDialog.Builder delegate;
+    private final String statePrefix;
+    private final Bundle state;
+
+    /**
+     * Create a new dialog builder with state tracking enabled.
+     *
+     * @param context is used for {@code AlertDialog.Builder}
+     * @param statePrefix as prefix for all keys of the state
+     * @param state is used to store the {@code TextView} texts
+     */
+    StateTrackingDialogBuilder(Context context, String statePrefix, Bundle state) {
+        delegate = new AlertDialog.Builder(context);
+        this.statePrefix = statePrefix;
+        this.state = state;
+    }
+
+    /**
+     * Calls {@link AlertDialog.Builder#setView(View)} under the hood and tracks state of all
+     * {@code TextView} children.
+
+     * @param view the view of the dialog
+     * @return builder instance for chaining
+     */
+    StateTrackingDialogBuilder setView(View view) {
+        delegate.setView(view);
+
+        if (view instanceof ViewGroup) {
+            ViewGroup viewGroup = (ViewGroup) view;
+            for (int i = 0; i < viewGroup.getChildCount(); i++) {
+                View child = viewGroup.getChildAt(i);
+
+                if (child instanceof TextView) {
+                    trackState((TextView) child, statePrefix + child.getId());
+                }
+            }
+        }
+
+        return this;
+    }
+
+    /**
+     * Tracks state of the {@code textView}. That includes initial state setting as well change
+     * detection in case of UI changes.
+     *
+     * @param textView to track state
+     * @param stateId key to use for the state bundle that was passed in the constructor
+     */
+    private void trackState(TextView textView, String stateId) {
+        String stateTextValue = state.getString(stateId);
+        if (stateTextValue != null) {
+            textView.setText(stateTextValue);
+        } else {
+            state.putString(stateId, textView.getText().toString());
```

Comment:
Now getting the initial state from textView. This resolves https://github.com/ruleant/getback_gps/pull/197#discussion_r810691721 and is the way you suggested in https://github.com/ruleant/getback_gps/pull/197#discussion_r810693548.